source/common/network/socket_option_impl.h

```diff
@@ -84,23 +84,31 @@ typedef absl::optional<std::pair<int, int>> SocketOptionName;
 
 class SocketOptionImpl : public Socket::Option, Logger::Loggable<Logger::Id::connection> {
 public:
-  SocketOptionImpl(Socket::SocketState in_state, Network::SocketOptionName optname, int value)
+  SocketOptionImpl(envoy::api::v2::core::SocketOption::SocketState in_state,
+                   Network::SocketOptionName optname, int value)
```

Comment:
do we really want to use 'int' here?  Not int32_t?

Just a comment here about why this an unspecified size is enough.

Reply:
From the man page for setsockopt:

> Most socket-level options utilize an int argument for optval.

Same for most socket options that are either a number or a boolean. So 'int' is definitely correct for the call to setsockopt. But possibly a different type would appropriate for the interface? I'm not sure. Comments about this are a good idea though.

Reply:
I'm already uncomfortable with the fact that the proto def is int64 and we stuff that into an unspecified sized type, but I agree that it's the right "wire" format for machines with 64 bit ints.

But given that the sockopt interfaces are defined in terms of ints, I'd like to normalize the types as early as possible.